synapsis/src/main/java/com/soen/synapsis/appuser/AppUser.java

```diff
@@ -118,12 +122,34 @@ public void setAuthProvider(AuthProvider authProvider) {
         this.authProvider = authProvider;
     }
 
-    public Long getCompanyId() {
-        return companyId;
+    public Set<AppUser> getRecruiter() {
+        if(this.getRole() != Role.COMPANY) {
+            throw new IllegalStateException("You must be company to have recruiters.");
+        }
+        return recruiters;
+    }
+
+    public void setRecruiter(AppUser recruiter) {
+        if(this.getRole() != Role.COMPANY) {
+            throw new IllegalStateException("You must be a company to set recruiters.");
+        }
+        if(this.recruiters == null) {
+            this.recruiters = new HashSet<AppUser>();
+        }
+        if(recruiter.getRole() == Role.RECRUITER) {
+            recruiters.add(recruiter);
+        }
+        if(recruiter.getRole() != Role.RECRUITER) {
+            recruiters.remove(recruiter);
+        }
+    }
+
+    public AppUser getCompany() {
+        return company;
     }
 
-    public void setCompanyId(Long companyId) {
-        this.companyId = companyId;
+    public void setCompany(AppUser company) {
```

Comment:
You should do a recruiter check here to since only recruiters can have the company field as non-null

Reply:
Should be good now.